### PR TITLE
Propagate the size of the file when declaring bad file : Closes #1610

### LINKED ISSUE #1610: Propagate the size of the file when declaring bad file
Motivation
----------
The `bad_replicas` table was extended with a bytes column for monitoring purpose. This column needs to be populated when a bad file is declared

### PR DESCRIPTION
**Pull request by cserf**
Propagate the size of the file when declaring bad file : Closes #1610

### CODE REVIEW

File: lib/rucio/core/replica.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def __exists_replicas(rse_id, scope=None, name=None, path=None, session=None):
             path_clause.append(models.RSEFileAssociation.path == path[1:])
         else:
             path_clause.append(models.RSEFileAssociation.path == '/%s' % path)
-        query = session.query(models.RSEFileAssociation.path, models.RSEFileAssociation.scope, models.RSEFileAssociation.name, models.RSEFileAssociation.rse_id).\
+        query = session.query(models.RSEFileAssociation.path, models.RSEFileAssociation.scope, models.RSEFileAssociation.name, models.RSEFileAssociation.rse_id, models.RSEFileAssociation.bytes).\
```
Comment (vingar) [inline lib/rucio/core/replica.py:118]: I would suggest to break this long line in several to improve readability.

**Review by vingar — COMMENTED** [review]
One non blocking request. Otherwise looks Ok.

File: lib/rucio/core/replica.py [hunk 2]
```diff
@@ -906,7 +906,7 @@ def _list_replicas(dataset_clause, file_clause, state_clause, show_pfns,
                                 # is the RSE site-configured?
                                 rse_site_attr = get_rse_attribute('site', rse_info[rse]['id'], session=session)
                                 replica_site = ['']
-                                if isinstance(rse_site_attr, list) and len(rse_site_attr):
```
Comment (mlassnig) [inline lib/rucio/core/replica.py:909]: this actually checks if there is an item in the list, you can't remove the `len` check

Comment (cserf) [reply lib/rucio/core/replica.py:909]: ??
The test is the same
>>> rse_site_attr = []
>>> rse_site_attr is False
False
>>> len(rse_site_attr) is False
False

Comment (mlassnig) [reply lib/rucio/core/replica.py:909]: aaargg, the original statement is wrong even that's why i got confused.
it really should be ... `and len(rse_site_attr)>0`

Comment (cserf) [reply lib/rucio/core/replica.py:909]: ???

`if len(rse_site_attr)>0` or `if len(rse_site_attr)` or `if rse_site_attr` should give the same result

**Review by mlassnig — APPROVED** [review]

**Pull request merged**